src/linagora.esn.unifiedinbox/app/components/composer/body-editor/html/composer-body-editor-html.spec.js

```diff
@@ -146,6 +153,31 @@ describe('The inboxComposerBodyEditorHtml component', function() {
 
       expect($rootScope.message.htmlBody).to.equal('<p><br></p><div class="openpaas-signature">-- \nmy signature</div>');
     });
+
+    it('should call sanitizeStylisedHtml filter', function() {
+      compileComponent();
+      const unsafeHtml = '<script>alert("hi");</script><h1>hello</h1><img src=x onerror=alert(1)//>';
+      const onPasteEvent = {
+        preventDefault: () => {},
+        clipboardData: {
+          getData: () => unsafeHtml
+        }
+      };
+
+      controller.onSummernotePaste(onPasteEvent);
+      $rootScope.$digest();
+
+      const trustedValue = $filter('sanitizeStylisedHtml')(unsafeHtml).$$unwrapTrustedValue();
+
+      expect(trustedValue).to.equal('<h1>hello</h1><img src="x" />');
```

Comment:
should be something like:
```js
expect(document.execCommand).to.have.been.calledWith('insertHtml', false, '<h1>hello</h1><img src="x" />');
```

Reply:
I have already tested this, but it does not work to mock `document`